src/_P090_MitsubishiHeatPump .ino

```diff
@@ -0,0 +1,698 @@
+static const byte FAN[6]              = {0x00,  0x01,   0x02, 0x03, 0x05, 0x06};
+static const char* FAN_MAP[6]         = {"AUTO", "QUIET", "1", "2", "3", "4"};
+static const byte VANE[7]             = {0x00,  0x01, 0x02, 0x03, 0x04, 0x05, 0x07};
+static const char* VANE_MAP[7]        = {"AUTO", "1", "2", "3", "4", "5", "SWING"};
+static const byte WIDEVANE[7]         = {0x01, 0x02, 0x03, 0x04, 0x05, 0x08, 0x0c};
+static const char* WIDEVANE_MAP[7]    = {"<<", "<",  "|",  ">",  ">>", "<>", "SWING"};
+static const byte ROOM_TEMP[32]       = {0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+                                  0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f};
+static const int ROOM_TEMP_MAP[32]    = {10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25,
+                                  26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41};
+//static const byte TIMER_MODE[4]       = {0x00,  0x01,  0x02, 0x03};
+//static const char* TIMER_MODE_MAP[4]  = {"NONE", "OFF", "ON", "BOTH"};
+
+// HeatPump.cpp
+const char* lookupByteMapValue(const char* valuesMap[], const byte byteMap[], int len, byte byteValue) {
+  for (int i = 0; i < len; i++) {
+    if (byteMap[i] == byteValue) {
+      return valuesMap[i];
+    }
+  }
+  return valuesMap[0];
+}
+
+int lookupByteMapValue(const int valuesMap[], const byte byteMap[], int len, byte byteValue) {
+  for (int i = 0; i < len; i++) {
+    if (byteMap[i] == byteValue) {
+      return valuesMap[i];
+    }
+  }
+  return valuesMap[0];
+}
+
+int lookupByteMapIndex(const int valuesMap[], int len, int lookupValue) {
+  for (int i = 0; i < len; i++) {
+    if (valuesMap[i] == lookupValue) {
+      return i;
+    }
+  }
+  return -1;
+}
+
+int lookupByteMapIndex(const char* valuesMap[], int len, const char* lookupValue) {
+  for (int i = 0; i < len; i++) {
+    if (strcasecmp(valuesMap[i], lookupValue) == 0) {
+      return i;
+    }
+  }
+  return -1;
+}
+
+bool lookupValue(const char* valuesMap[], int len, const char* lookupValue, const char*& destination) {
+  int index = lookupByteMapIndex(valuesMap, len, lookupValue);
+  if (index == -1) {
+    return false;
+  }
+  destination = valuesMap[index];
+  return true;
+}
+
+byte checkSum(byte bytes[], int len) {
+  byte sum = 0;
+  for (int i = 0; i < len; i++) {
+    sum += bytes[i];
+  }
+  return (0xfc - sum) & 0xff;
+}
+
+static bool operator!=(const heatpumpSettings& lhs, const heatpumpSettings& rhs) {
+  return lhs.power != rhs.power ||
+         lhs.mode != rhs.mode ||
+         lhs.temperature != rhs.temperature ||
+         lhs.fan != rhs.fan ||
+         lhs.vane != rhs.vane ||
+         lhs.wideVane != rhs.wideVane ||
+         lhs.iSee != rhs.iSee;
+}
+
+P090_data_struct::P090_data_struct(const int16_t serialRx, const int16_t serialTx) :
+  _serial(serialRx, serialTx),
+  _currentSettings({}),
+  _currentStatus({}),
```

Comment:
`_currentSettings` and `_currentStatus` should have their own constructor which must set their members to proper defaults. Not sure what this does, but I doubt it will set all members to their binary "0" equivalent.